### PR TITLE
feat(convert): Convert Deepnote projects to Jupyter notebooks

### PR DESCRIPTION
**Pull request by Artmann**
```
npx @deepnote/convert project.deepnote
```

## Summary by CodeRabbit

* **New Features**
  * Bidirectional conversion between Deepnote and Jupyter notebooks with CLI support for .deepnote → .ipynb and lossless round-trip metadata preservation.
* **Documentation**
  * Expanded README with bidirectional usage examples, updated CLI options, output-path semantics, and a Lossless Roundtrip section.
* **Tests**
  * Comprehensive tests and fixtures covering file-based and in-memory conversions, round-trip integrity, diverse block types, filename/path edge cases, and error handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

## Summary by CodeRabbit

* **New Features**
  * Bidirectional conversion between Deepnote projects and Jupyter notebooks; CLI now supports Deepnote→Jupyter with output directory control.

* **Documentation**
  * README expanded with dual-direction usage, explicit commands, lossless roundtrip notes, and programmatic examples.

* **Tests**
  * Comprehensive tests added for Deepnote↔Jupyter conversions, error cases, filename/path handling, and roundtrip integrity.

* **Fixtures**
  * New rich notebook and Deepnote fixtures covering diverse block types, outputs, and interactive widgets.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/convert/src/cli.test.ts [hunk 1]
```diff
@@ -213,26 +214,69 @@ describe('CLI convert function', () => {
...
+
+    await fs.writeFile(deepnotePath, deepnoteContent, 'utf-8')
+
+    const outputDir = await convert({
+      inputPath: deepnotePath,
+      cwd: tempDir,
+    })
+
+    // Should derive output directory from input filename (test.deepnote -> test/)
+    const expectedOutputDir = path.join(tempDir, 'test')
```
Comment (saltenasl) [inline packages/convert/src/cli.test.ts]: I'd extract hardcoded `test.deepnote` in line 218 (`const deepnotePath = path.join(tempDir, 'test.deepnote')`) to a separate const `projectName` and the nuse it in the assertion. Now it's not immediately apparent why `'test'` is hardcoded here.

```suggestion
    const expectedOutputDir = path.join(tempDir, projectName)
```

File: packages/convert/src/cli.test.ts [hunk 2]
```diff
@@ -213,26 +214,69 @@ describe('CLI convert function', () => {
...
+    expect(outputDir).toBe(expectedOutputDir)
+
+    // Should create output directory with Jupyter notebook(s)
+    const stat = await fs.stat(outputDir)
+    expect(stat.isDirectory()).toBe(true)
+
+    const files = await fs.readdir(outputDir)
+    expect(files.length).toBeGreaterThan(0)
```
Comment (saltenasl) [inline packages/convert/src/cli.test.ts]: there's exactly 1 notebook, do we need to be safe asserting greater than 0?

File: packages/convert/src/cli.test.ts [hunk 3]
```diff
@@ -213,26 +214,69 @@ describe('CLI convert function', () => {
...
+    expect(files.some(file => file.endsWith('.ipynb'))).toBe(true)
```
Comment (saltenasl) [inline packages/convert/src/cli.test.ts]: ditto here, can we assert that the filename is `notebookName + '.ipynb'` or something along these lines?

File: packages/convert/src/roundtrip.test.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    for (let i = 0; i < original.project.notebooks.length; i++) {
+      const originalNotebook = original.project.notebooks[i]
+      const roundtrippedNotebook = roundtripped.project.notebooks[i]
+
+      expect(roundtrippedNotebook.executionMode).toBe(originalNotebook.executionMode)
+      expect(roundtrippedNotebook.isModule).toBe(originalNotebook.isModule)
+      expect(roundtrippedNotebook.workingDirectory).toBe(originalNotebook.workingDirectory)
+    }
+  })
+})
```
Comment (saltenasl) [inline packages/convert/src/roundtrip.test.ts:110]: can we also add Jupyter -> Deepnote -> Jupyter roundtrip? 

I think separately (nit) above all can be nested under `describe('Deepnote -> Jupyter -> Deepnote'` block

**Review by saltenasl — CHANGES_REQUESTED** [review]
I'd like a roundtrip Jupyter -> Deepnote -> Jupyter coverage, rest are nits

File: packages/convert/src/cli.test.ts (hunk 1 above)
Comment (Artmann) [reply packages/convert/src/cli.test.ts]: Fixed

File: packages/convert/src/cli.test.ts (hunk 2 above)
Comment (Artmann) [reply packages/convert/src/cli.test.ts]: Replaced with a stronger assertion

File: packages/convert/src/cli.test.ts (hunk 3 above)
Comment (Artmann) [reply packages/convert/src/cli.test.ts]: Replaced with a stronger assertion

File: packages/convert/src/roundtrip.test.ts (hunk 4 above)
Comment (Artmann) [reply packages/convert/src/roundtrip.test.ts:110]: Added

**Comment by Artmann** [comment]
> Thanks!

Thansk for the review <3
